client/src/app/site/motions/modules/motion-list/components/motion-list/motion-list.component.ts

```diff
@@ -396,6 +404,67 @@ export class MotionListComponent extends ListViewBaseComponent<ViewMotion, Motio
         };
     }
 
+    /**
+     * Function to get a list with all favorite motions.
+     *
+     * @returns An array containing all favorite motions.
+     */
+    public motionsAsFavorite(): ViewMotion[] {
+        return this.motionRepo.getSortedViewModelList().filter(motion => motion.star);
+    }
+
+    /**
+     * Function to get a list of contained motions in a category.
+     *
+     * @param category The category
+     *
+     * @returns An array with all the motions in the given category.
+     */
+    public motionsInCategory(category: Category): ViewMotion[] {
+        return this.motionRepo.getSortedViewModelList().filter(motion => motion.category_id === category.id);
+    }
+
+    /**
+     * Function to get the favourite view for the user.
+     */
+    private async initializeView(): Promise<void> {
+        this.selectedView = (await this.storage.get<string>('motionListView')) || 'tiles';
+    }
+
+    /**
+     * This function saves the selected view by changes.
+     *
+     * @param value is the new view the user has selected.
+     */
+    public onChangeView(value: string): void {
+        this.selectedView = value;
+        this.storage.set('motionListView', value);
+    }
+
+    /**
+     * This function navigates to the list view where the selected category becomes the active filter.
+     *
+     * @param event The event from the card the user selected. Contains the source and the data as `ViewCategory`.
+     */
+    public navigateTo(event: EventObject, toFavorites: boolean): void {
```

Comment:
split this into: `changeToListWithFavorites` and `changeToListWithCategory`. Use `change` instead of `navigate` to make it clear, that no navigation will be triggered. In the template you can just use `changeToListWithCategory(category)` without using `$event`.